cli/commands/scaffold/action.go

```diff
@@ -96,6 +101,21 @@ inputs = {
 var moduleURLRegex = regexp.MustCompile(moduleURLPattern)
 
 func Run(ctx context.Context, opts *options.TerragruntOptions, moduleURL, templateURL string) error {
+	if opts.ScaffoldRootFileName == "" {
```

Comment:
Duplicated code. If someone else works on this code, he most likely won't know that changes need to be synchronized in several places. I would recommend keeping this code in one place and just referencing it, or even better moving it into a scaffold.RootFileName flag and defining it in `commands/flags.go`

```go
func NewRootFileNameFlag(opts *options.TerragruntOptions) cli.Flag {
	return &cli.GenericFlag[string]{
			Name:        scaffold.RootFileName,
			Destination: &opts.ScaffoldRootFileName,
			Usage:       "Name of the root Terragrunt configuration file, if used.",
			Action: func(ctx *cli.Context, value string) error {
				if value == config.DefaultTerragruntConfigPath {
					if control, ok := strict.GetStrictControl(strict.RootTerragruntHCL); ok {
						warn, triggered, err := control.Evaluate(opts)
						if err != nil {
							return err
						}

						if !triggered {
							opts.Logger.Warnf(warn)
						}
					}
				}

				return nil
			},
		},
}
```

them in both places scaffold/command.go and catalog/command.go just referencing to that func

```go
func NewFlags(opts *options.TerragruntOptions) cli.Flags {
	return cli.Flags{
		commands.NewRootFileNameFlag(opts),
                ....
	}
}
```